content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+.. Name: *HTTP_FORWARDED*
+. Click on _Back to Rules_
+. _Click on Add Rule(s)..._
+. Select _Reverse Proxy_ and click on OK
+. Enter _jenkins.acme.example_ and click on OK
+. Open the rule you just created
+. Under _Conditions_ add:
+.. Condition input: *\{CACHE_URL}*
+.. Pattern: *^(http|ws)s://*
+. Under _Server Variables_ add:
+.. Name: *HTTP_FORWARDED*, Value:
+*for=\{REMOTE_ADDR};by=\{LOCAL_ADDR};host="\{HTTP_HOST}";proto="https"*,
+Replace: yes
+... Jenkins runs under Jetty, Jetty supports
+https://tools.ietf.org/html/rfc7239[RFC7239], so all should be well.
+. Under Action change:
+..  Rewrite URL to
+*\{C:1}[.nolink]#://jenkins.acme.example:8080\{UNENCODED_URL#}*
+... Note that there is no slash between the port number and the opening
+curly bracket
+.. *Remove* the check from the *Append query string* checkbox
+. Apply the changes.
+. Edit _C:\Windows\System32\drivers\etc\hosts_ so that
+*jenkins.acme.example* points to 127.0.0.1
+.. When resolving names Windows will check if the name is its own name
+before consulting the hosts file. Meaning that adding _vrtjenkins01_ or
+_vrtjenkins01.acme.example_ to the hosts file won't have any effect.
+... The hosts file will however be consulted before consulting the DNS
+infrastructure
+
+===== Experiencing the dreaded "It appears that your reverse proxy set up is broken." error for yourself
+
+. [.nolink]#https://jenkins.acme.example/configure#
+. Configure the _Jenkins URL_ to
+be **[.nolink]#https://jenkins.acme.example/#**
+and save the change
+. Go to _Configure Global Security_ and enable
+_Enable proxy compatibility_ if you have already enabled _Prevent Cross
+Site Request Forgery exploits_
+. Go to [.nolink]#https://jenkins.acme.example/manage#
+. You will still experience the "It appears that your reverse
+proxy set up is broken." as expected
+.. If you do not get that at this point, then that is very weird...
+Continue anyway.
+. Right click the _Configure System_ link and choose to
+inspect the element.
+.. Make sure you are still on the Manage page as you will want
+it as your referrer
+. Change the value of the _href_ attribute to be
+_administrativeMonitor/hudson.diagnosis.ReverseProxySetupMonitor/test_
+. Open the link you just changed in a new tab.
+.. Keep this tab open
+. Observe the "[.nolink]#https://jenkins.acme.example/manage#
+vs http:" error and bask in its glory
+.. a white page served with HTTP status code is 200 indicates
+all is well
+... If you do get that at this point, then that is very
+weird... Continue anyway.
+
+===== [.nolink]#Fixing the errors#
+
+. In IIS Manager got to _Application Pools_ then edit
+_DefaultAppPool_ so that the _.NET CLR version_ is *No Managed Code*
+.. You might find that this is not necessary (at far as you
+can tell) for your setup, since IIS will only act as a TLS/SSL
+offloading reverse proxy, we don't need it. IIS on Windows 7 does appear
+to need this to be turned off for some urls
+(https://jenkins.acme.example/*)
```

Comment:
No need to describe IIS on Windows 7.  Windows 7 support ended January 14, 2020.
```suggestion
offloading reverse proxy, we don't need it.
```